src/main/webapp/app/entities/participation/participation.model.ts

```diff
@@ -49,7 +49,12 @@ export abstract class Participation implements BaseEntity {
     }
 }
 
-export const getExercise = (participation: Participation): Exercise | undefined => {
+/**
+ * Gets the exercise of a given participation, if possible.
+ *
+ * This is needed as different participation types may use different exercise attributes (e.g. not 'exercise' but 'programmingExercise').
+ */
+export const getExercise = (participation: Participation | undefined): Exercise | undefined => {
```

Comment:
I don't understand why this should be undefined-able. If I ask for the exercise of undefined it should be trivial that this is not possible to retrieve -> undefined should be captured before accessing this function.

Reply:
I thought it might simplify handling possibly undefined values, but you are right, I can revert this. I would also remove the `if` then.